### ISSUE TITLE
Lithuanian letters then I get task from mail it come with unknow simbols (?ra?o)

### ISSUE THREAD

**Issue by MindaugasP88** [glpi-project/glpi#4559]
Hello, 

Have any one face problem with task formed from email and letters come in weird way  ( ?ra?as i? ) it happen not from all users sent mail.  
![paveikslas](https://user-images.githubusercontent.com/42863157/44904644-7d6e4880-ad18-11e8-9161-542c05761ab2.png)

**Comment by trasher** [comment]
We've already seen some issues due to software used to send the email. What are faultly usrs using as mail client?

**Comment by MindaugasP88** [comment]
From  Mozila thunderbird , apple, outlook 2016 and  phone clients do jab nice letters comes as it must by . With outlook 2007and outlook 2013 it print in "????????" way  
![paveikslas](https://user-images.githubusercontent.com/42863157/44987240-30e07280-af8f-11e8-82a5-3b56595e2003.png)

**Comment by trasher** [comment]
Thank you for the feedback. It seems there are many encoding issues with outlook; we do not know why exactly... I do not use outlook at all (I do not run windows either); so I can't help a lot.

**Comment by cedric-anne** [comment]
Hi,

I do not use Outlook, so I cannot make a real test. I take a look on the conversion process, and it may fail if there is no `CHARSET` parameter found in `imap_fetchstructure`.

First solution could be to fallback to detected charset using this patch :
```diff
diff --git a/inc/mailcollector.class.php b/inc/mailcollector.class.php
index 48f63d027..80aeca5e4 100644
--- a/inc/mailcollector.class.php
+++ b/inc/mailcollector.class.php
@@ -1396,17 +1396,26 @@ class MailCollector  extends CommonDBTM {
                $text = str_replace("\n", " ", $text);
             }
 
-            if (count($structure->parameters) > 0) {
-               foreach ($structure->parameters as $param) {
-                  if ((strtoupper($param->attribute) == 'CHARSET')
-                      && function_exists('mb_convert_encoding')
-                      && (strtoupper($param->value) != 'UTF-8')) {
+            if (function_exists('mb_convert_encoding')) {
+               $detected_charset = null;
 
-                     $text                 = mb_convert_encoding($text, 'utf-8', $param->value);
-                     $this->body_converted = true;
+               foreach ($structure->parameters as $param) {
+                  if ((strtoupper($param->attribute) == 'CHARSET')) {
+                     $detected_charset = $param->value;
+                     break;
                   }
                }
+
+               if (null === $detected_charset) {
+                  $detected_charset = mb_detect_encoding($text, null, true);
+               }
+
+               if ($detected_charset && strtoupper($detected_charset) != 'UTF-8') {
+                  $text                 = mb_convert_encoding($text, 'utf-8', $detected_charset);
+                  $this->body_converted = true;
+               }
             }
+
             return $text;
          }
```

Second solution could be fixing fallback conversion using this patch :
```diff
diff --git a/inc/mailcollector.class.php b/inc/mailcollector.class.php
index 48f63d027..29598c080 100644
--- a/inc/mailcollector.class.php
+++ b/inc/mailcollector.class.php
@@ -849,7 +849,7 @@ class MailCollector  extends CommonDBTM {
       }
 
       if (!Toolbox::seems_utf8($body)) {
-         $tkt['content'] = Toolbox::encodeInUtf8($body);
+         $tkt['content'] = Toolbox::encodeInUtf8($body, 'auto');
       } else {
          $tkt['content'] = $body;
       }
```

@MindaugasP88 could you test this ?

Regards

**Comment by klodnitsky** [comment]
GLPI 9.3.1, Latvian language, same issue.

@cedric-anne with your patches same thing when sending e-mail from Outlook 16 and Baltic (Windows) encoding. So... problem exists.

Working correctly only when in outlook encoding set to utf8

**Comment by cedric-anne** [comment]
OK, so I guess we could consider it is a bug. I have no time to investigate right now, but I will do it maybe next week.

**Comment by cedric-anne** [comment]
Hi,

Sorry for the late.

Could you test the fix provided in #4844 ?

Regards

**Comment by cedric-anne** [comment]
@MindaugasP88 @klodnitsky ,

Could you test fix provided in #4844 ?

We would prefer to have a feedback before merging the fix.

Regards

**Comment by trasher** [comment]
No feedback from a while, closing.
